modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/utils/SW360ComponentAdapterUtils.java

```diff
@@ -45,7 +46,7 @@ public static String createComponentName(Artifact artifact) {
         return getMostDominantArtifactCoordinates(artifact)
                 .map(ArtifactCoordinates::getName)
                 .filter(n -> ! n.isEmpty())
-                .orElse(artifact.toString()); // TODO: ugly hack
+                .orElseThrow(() -> new ExecutionException("Component name of " + artifact.toString() + " could not be created"));
```

Comment:
I do not think, that this is a case for an Exception, no coordinates is not really very likely, since we try to handle this in most cases with generic coordinates. But if this is the case for whatever reason, the result can be something ugly, but processing should continue.

Reply:
I referred to the TODO statement :-)